pandas/tests/sparse/frame/test_frame.py

```diff
@@ -350,6 +350,18 @@ def test_dense_to_sparse(self):
         assert sdf.default_fill_value == 0
         tm.assert_frame_equal(sdf.to_dense(), df)
 
+    def test_deprecated_dense_to_sparse(self):
+        # GH 26557
+        # Deprecated 0.25.0
+
+        df = pd.DataFrame({"A": [1, np.nan, 3]})
+        sparse_df = pd.SparseDataFrame({"A": [1, np.nan, 3]})
+
+        with tm.assert_produces_warning(FutureWarning,
+                                        check_stacklevel=False):
+            result = df.to_sparse()
+        tm.assert_frame_equal(result, sparse_df)
```

Comment:
you put this in a class where those warnings you want to check for are generally ignored. Does that work? (just wondering)

Maybe we can also put it at the end of the file as a function instead of method, to avoid possible concerns about that.

Reply:
But since the tests are passing, this is probably fine ;)

Reply:
This actually works fine, `tm.assert_produces_warning` doesn't seem to care about filterwarnings. Remove the warning from `to_sparse` and this test fails. 

I wanted to keep both (`test_dense_to_sparse` and `test_deprecated_dense_to_sparse`) close for easy reference, thats all. Open to shifting it to the end though.

Reply:
Thanks for checking, then that sounds good!